src/datumaro/plugins/ovms_plugin/launcher.py

```diff
@@ -0,0 +1,191 @@
+        timeout: float = 10.0,
+        tls_config: Optional[TLSConfig] = None,
+        ovms_client_type: OVMSClientType = OVMSClientType.grpc,
+    ):
+        super().__init__(model_interpreter_path=model_interpreter_path)
+
+        self._client = self._init_client(
+            model_name,
+            host,
+            port,
+            tls_config,
+            ovms_client_type,
+        )
+        self._check_server_health(model_version, timeout)
+        self._init_input_name(model_version, timeout)
+
+        self.model_version = model_version
+        self.timeout = timeout
+
+    def _init_client(
+        self,
+        model_name,
+        host,
+        port,
+        tls_config,
+        ovms_client_type,
+    ) -> Union[GrpcClient, HttpClient]:
+        self.model_name = model_name
+        self.url = f"{host}:{port}"
+        self.tls_config = tls_config
+
+        if ovms_client_type == OVMSClientType.grpc:
+            return make_grpc_client(self.url, self.tls_config)
+        elif ovms_client_type == OVMSClientType.http:
+            return make_http_client(self.url, self.tls_config)
+        else:
+            raise NotImplementedError(ovms_client_type)
+
+    def _check_server_health(self, model_version, timeout):
+        try:
+            status = self._client.get_model_status(
+                model_name=self.model_name,
+                model_version=model_version,
+                timeout=timeout,
+            )
+            log.info(f"Health check succeeded: {status}")
+        except Exception as e:
+            raise DatumaroError(
+                f"Health check failed for model_name={self.model_name}, "
+                f"model_version={model_version}, url={self.url} and tls_config={self.tls_config}"
+            ) from e
+
+    def _init_input_name(self, model_version, timeout):
+        metadata = self._client.get_model_metadata(
+            model_name=self.model_name,
+            model_version=model_version,
+            timeout=timeout,
+        )
+        metadata_inputs = metadata.get("inputs")
+        if metadata_inputs is None:
+            raise DatumaroError("Cannot get metadata of the inputs.")
+
+        if len(metadata_inputs.keys()) > 1:
+            raise DatumaroError(
+                f"More than two model inputs are not allowed: {metadata_inputs.keys()}."
+            )
+
+        self._input_key = next(iter(metadata_inputs.keys()))
+        log.info(f"Model input key is {self._input_key}")
+
+    def infer(self, inputs: np.ndarray) -> List[ModelPred]:
+        results = self._client.predict(
+            inputs={self._input_key: inputs},
+            model_name=self.model_name,
+            model_version=self.model_version,
+            timeout=self.timeout,
+        )
+
+        # If there is only one output key,
+        # it returns `np.ndarray`` rather than `Dict[str, np.ndarray]`.
+        # Please see ovmsclient.tfs_compat.grpc.responses.GrpcPredictResponse
```

Comment:
It seems good to add [GrpcClient.predict](https://github.com/openvinotoolkit/model_server/blob/releases/2022/3/client/python/ovmsclient/lib/docs/grpc_client.md#method-predict) in this comment or add link of this in documentation for this feature.

Reply:
Updated in https://github.com/openvinotoolkit/datumaro/pull/1056/commits/afd0b4007e96c26de2d43e3d5df1f4c2a78d1799